### PR TITLE
Add a change log to the documentation.

### PR DESCRIPTION
**Pull request by funkaoshi**
A customer asked us to provide them with a list of what changes (in more
detail) from release to release. This is probably a good idea to
maintain. I'm sure it will be a common request. Will also help us track
our own progress on the API, and when changes were made.

### CODE REVIEW

File: source/includes/changelog.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+## v4.2
+
+- Add documentation for session and basic authentication.
+- Business Units, Groups, and Tasks includes the role of the users the resource returns.
+- Include the first name, last name for the users returned by the tasks endpoint.
+- ALM and Analysis Connections include whether they are accessible or not, and the id of their parent Connector. 
+- ALM and Analysis Connections can be created, edited, or deleted.
+- ALM sync jobs can be initiated by POSTing to the alm jobs endpoint.
+- Analysis import jobs can be initiated by POSTing to the analysis jobs endpoint.
+-  Project Role can be ordered and searched for by name.
```
Comment (funkaoshi) [inline source/includes/changelog.md]: I'll fix the weird double space here.

**Comment by geoffwhittington** [comment]
+1

File: README.md [hunk 2]
```diff
@@ -2,15 +2,15 @@
 
 [![Build Status](https://travis-ci.org/sdelements/slate.svg?branch=master)](https://travis-ci.org/sdelements/slate)
 
-### Set Up
+## Set Up
```
Comment (sibartlett) [inline README.md]: Should maybe stick with 3 #'s for time-being, to reduce conflicts with upstream?

Comment (sibartlett) [inline README.md]: What am I saying? Ignore that, we've completely changed the README compared to upstream. hah

Comment (funkaoshi) [inline README.md]: Ha. Yeah, I wasn't sure if there was a reason it was 3 instead of 2.

**Pull request merged**